etc/gen_atlas_search/src/main.rs

```diff
@@ -0,0 +1,219 @@
+use convert_case::{Case, Casing};
+use proc_macro2::TokenStream;
+use quote::format_ident;
+use serde::Deserialize;
+use syn::parse_quote;
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+struct Operator {
+    name: String,
+    #[expect(dead_code)]
+    link: String,
+    #[serde(rename = "type")]
+    #[expect(dead_code)]
+    type_: Vec<OperatorType>,
+    #[expect(dead_code)]
+    encode: EncodeType,
+    description: String,
+    arguments: Vec<Argument>,
+    tests: Vec<serde_yaml::Value>,
```

Comment:
The yaml file test definitions aren't terribly useful here, so I just cheated the parsing.